bodhi/tests/server/functional/test_updates.py

```diff
@@ -2520,6 +2520,99 @@ def test_edit_testing_update_reset_karma_with_same_tester(self, publish, *args):
 
     @mock.patch(**mock_valid_requirements)
     @mock.patch('bodhi.server.notifications.publish')
+    def test_has_negative_karma(self, publish, *args):
+        """
+        The test asserts True for has_negative_karma i.e,
+        if update has negative karma
+        """
+        user = User(name=u'bob')
+        self.db.add(user)
+        self.db.flush()
+
+        nvr = 'bodhi-2.1-1.fc17'
+        args = self.get_update(nvr)
+        resp = self.app.post_json('/updates/', args).json_body
+        publish.assert_called_with(topic='update.request.testing', msg=ANY)
+
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        up.request = None
+        up.status = UpdateStatus.testing
+        self.db.flush()
+
+        # The user gives negative karma first
+        up.comment(self.db, u'Failed to work', author=u'ralph', karma=-1)
+        up = self.db.query(Update).filter_by(title=nvr).one()
+        self.assertTrue(up.has_negative_karma)
```

Comment:
It looks like your ```test_has_no_negative_karma``` test below does test this condition. You can drop this assertion to make this test a little bit more readable and cleaner, since the other test already makes sure this works.